frontend/app/controllers/spree/checkout_controller.rb

```diff
@@ -124,6 +125,18 @@ def ensure_valid_state
       end
     end
 
+    def ensure_store_credits_eligible
+      expired_payments = @order.payments.valid.store_credits.select { |p| p.source.expired? }
```

Comment:
I discussed this with @cbrunsdon offline because I initially had this in a `before_transition to: :complete`. He brought up that there would likely be scenarios where admins would want to push through an order with expired credit and having it in a transition hook would prevent them from being able to do that. In general, having it at the model level would make it hard to opt out of this behaviour if desired.

I could also add something to the API checkout controller to ensure similar behaviour, or is there a place you prefer this sort of logic to live?